### DISCUSSION TITLE
Building a Component library

### DISCUSSION THREAD

**Discussion by VaZark** [django-components/django-components#186] [Q&A] [answered]
Just found this project and it really fills a niche that pushes back against the trend of isingthat django as a backend framework. Would’ve loved to have started a few recent projects with this.

My questions are, 

1. is there any way we can visualise and test the components designed with something like storybook?
2.  How do you handled adding global settings for libs like TailwindCSS? (global cdn?)
3. How are the media files (css/js) bundled? Are they rolled together like a component library and served as one single js/css file ?

I love the idea of bringing some of the React-style intuitiveness of components to template renderers

**Comment by EmilStenstrom** [comment, marked as answer]
Great questions. First of all django-components isn't really that big, so if there's something missing that you like, you could easily add it!

1. Visualizing components: All components are registered in a registry. I think you could loop over the registry and instanciate all components that way? If you need default values for the component values, you could maybe add a new method that provides these. Do you think that could work?
2. If you just want a global `<link>` or `<script>` tag, I wouldn't put them in a component at all. Instead, just put them in your global html-file and be done with it. 
3. This library actually does not handle bundling at all, it's left to a separate library. We use a subclass of Form.media here, which many bundlers support well. There's an open issue to [document the exact way to do this](https://github.com/EmilStenstrom/django-components/issues/39), feel free to help!

**Reply by VaZark** [reply]
Thanks for the rapid response.

1. I’ll probably play around to see how to grok Vanilla components on storybook over the week
2. Global cdns are pretty neat for most cases but some libs can passed through shaking to reduce bundle size. Which neatly segues to q3
3. Django-compressor looks neat. I’ll try to see if we can bundle and apply the beat of JS for smaller bundle sizes
